targets/posix/Include/nanoPAL.h

```
@@ -0,0 +1,53 @@
+#include <nanoPAL_Time.h>
+
+// hal_strlen_s - mapped to strlen on all non-VIRTUAL_DEVICE builds
+#include <string.h>
+#define hal_strlen_s(str) strlen(str)
+
+// hal_strcpy_s / hal_strncpy_s — Annex K not available on POSIX; map to safe equivalents
+// strncpy_s(dst, size, src, count): copy at most count chars, always NUL-terminate
+// We satisfy the contract: return 0 on success.
+inline int hal_strcpy_s(char *dst, size_t size, const char *src)
+{
+    if (!dst || size == 0)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)